### PR TITLE
Refactor dashboard tests

### PR DESCRIPTION
**Pull request by nmajor25**
## Summary by CodeRabbit

- **New Features**
	- Introduced a dynamic approach for dashboard testing, improving maintainability and flexibility.
	- Added new functions to retrieve and enhance dashboard data with unique identifiers.
	- Implemented a mock dashboard generator for realistic test data.

- **Bug Fixes**
	- Updated test logic to ensure correct dashboard ID references, improving test robustness.

### CODE REVIEW

File: test/support/api/dashboards.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-import { getResource } from 'helpers/json-api';
+import { v4 as uuid } from 'uuid';
+
+import { getResource, mergeJsonApi } from 'helpers/json-api';
 
 import fxTestDashboards from 'fixtures/test/dashboards';
 
 const TYPE = 'dashboards';
 
+export function getDashboard(data) {
```
Comment (nmajor25) [inline test/support/api/dashboards.js:10]: We could use `getDashboard()` in the `routeDashboards` command as well, I think? With a sample of `3`?

File: test/support/api/dashboards.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+  const resource = getResource(_.sample(fxTestDashboards), TYPE);
+
+  data = _.extend({ id: uuid() }, data);
+
+  return mergeJsonApi(resource, data);
+}
+
 Cypress.Commands.add('routeDashboards', (mutator = _.identity) => {
```
Comment (nmajor25) [inline test/support/api/dashboards.js:19]: We could use `getDashboard()` in the `routeDashboards` command as well, I think? With a sample of `3`?

Comment (paulfalgout) [reply test/support/api/dashboards.js:19]: I think that's probably correct

**Review by paulfalgout — CHANGES_REQUESTED** [review]
We should be able to move the dashboard fixtures to the generated ones

File: test/support/api/dashboards.js (hunk 2 above)
Comment (nmajor25) [reply test/support/api/dashboards.js:19]: Made this update ☝️

Comment (nmajor25) [reply test/support/api/dashboards.js:19]: It didn't work, reverted it back for now.

Comment (nmajor25) [reply test/support/api/dashboards.js:19]: Okay, I got some working for this. In latest commit.

File: test/support/api/dashboards.js [hunk 3]
```diff
@@ -1,12 +1,23 @@
 import _ from 'underscore';
-import { getResource } from 'helpers/json-api';
 
-import fxTestDashboards from 'fixtures/test/dashboards';
+import { getResource, mergeJsonApi } from 'helpers/json-api';
+
+import fxTestDashboards from 'fixtures/collections/dashboards';
```
Comment (nmajor25) [inline test/support/api/dashboards.js:5]: Moved these to a generated fixture file.

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**